packages/admin/cms-admin/src/userPermissions/user/permissions/ContentScopeGrid.tsx

```diff
@@ -56,57 +55,94 @@ export const ContentScopeGrid: React.FC<{
-            >
-                <CardToolbar>
-                    <ToolbarTitleItem>
-                        <FormattedMessage id="comet.userPermissions.scopes" defaultMessage="Scopes" />
-                    </ToolbarTitleItem>
-                    <ToolbarFillSpace />
-                    <ToolbarActions>
-                        <SaveButton type="submit">
-                            <FormattedMessage id="comet.userPermissions.save" defaultMessage="Save" />
-                        </SaveButton>
-                    </ToolbarActions>
-                </CardToolbar>
-                <CardContent>
-                    {data.availableContentScopes.map((contentScope: ContentScope) => (
-                        <Field
-                            disabled={data.userContentScopesSkipManual.some((cs: ContentScope) => isEqual(cs, contentScope))}
-                            key={JSON.stringify(contentScope)}
-                            name="contentScopes"
-                            fullWidth
-                            variant="horizontal"
-                            type="checkbox"
-                            component={FinalFormCheckbox}
-                            value={JSON.stringify(contentScope)}
-                            label={Object.entries(contentScope).map(([scope, value]) => (
-                                <>
-                                    {camelCaseToHumanReadable(scope)}: {camelCaseToHumanReadable(value)}
-                                    <br />
-                                </>
-                            ))}
-                        />
-                    ))}
-                </CardContent>
-            </FinalForm>
-        </Card>
+        <FieldSet
+            title={<FormattedMessage id="comet.userPermissions.scopes.assigned" defaultMessage="Assigned Scopes" />}
+            initiallyExpanded
+            disablePadding
+        >
+            <Card>
+                <DataGrid<GQLUserContentScopes>
+                    autoHeight
+                    checkboxSelection
+                    disableSelectionOnClick
+                    getRowId={(row) => {
+                        return JSON.stringify(row);
+                    }}
+                    selectionModel={selectionModel}
+                    onSelectionModelChange={(newSelectionModel) => {
+                        setSelectionModel(newSelectionModel);
+                    }}
+                    rows={data?.userContentScopes ?? []}
+                    columns={columns}
+                    rowCount={data?.userContentScopes.length ?? 0}
+                    loading={loading}
+                    getRowHeight={() => "auto"}
+                    sx={{ "&.MuiDataGrid-root .MuiDataGrid-cell": { py: "8px" } }}
+                    components={{
+                        Toolbar: () => <ContentScopeGridToolbar actions={{ openDialog: openSetContentScopesDialog, deleteSelected }} />,
+                        OpenFilterButtonIcon: () => <Filter />,
```

Comment:
Component props should be passed using componentsProps, right @jamesricky?

Reply:
Yes. 

`componentsProps`/`slotProps` should be used to override props of a child-component. 
`components`/`slots` should only be used to entirely replace that child-component.